### PR TITLE
test(core): add exhaustive decomposition test on several levels

### PR DESCRIPTION
**Pull request by IceTDrinker**
- also add a verification that recomposition works in the previously added edge case test

### CODE REVIEW

File: tfhe/src/core_crypto/commons/math/decomposition/tests.rs [hunk 1]
```diff
@@ -230,11 +230,49 @@ fn test_decomposition_edge_case_sign_handling() {
...
+#[test]
+fn test_recompose_exhaustive() {
+    let base_log = DecompositionBaseLog(10);
+    let level = DecompositionLevelCount(3);
+    let decomposer = SignedDecomposer::new(base_log, level);
+
+    assert!(
+        decomposer.level_count().0 > 1,
+        "This test expects more than 1 level in the decomposer"
+    );
```
Comment (JJ-hw) [inline tfhe/src/core_crypto/commons/math/decomposition/tests.rs:254]: Why is this restriction necessary ?
Considering the test range (u32), maybe the constraint is that `base * level` < 32.

Comment (IceTDrinker) [reply tfhe/src/core_crypto/commons/math/decomposition/tests.rs:254]: we have both, we already have a test for balanced decomp over one level (was a separate issue that we fixed a while ago, so here I want to make sure we have strictly more than one level)

you are right on the limitation you mention, I can add it to be safe
